spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkMLTrainer.java

```diff
@@ -0,0 +1,145 @@
+    protected String featureFieldsToExclude;
+
+    @Description("The field from which to get the label. It must be of type double.")
+    protected String labelField;
+
+    protected MLTrainerConfig() {
+      System.out.print("hello");
+    }
+
+    protected MLTrainerConfig(String fileSetName, String path, @Nullable String featureFieldsToInclude,
+                              @Nullable String featureFieldsToExclude, String labelField) {
+      this.fileSetName = fileSetName;
+      this.path = path;
+      this.featureFieldsToInclude = featureFieldsToInclude;
+      this.featureFieldsToExclude = featureFieldsToExclude;
+      this.labelField = labelField;
+    }
+
+    public void validate(Schema inputSchema) {
+      SparkUtils.validateConfigParameters(inputSchema, featureFieldsToInclude, featureFieldsToExclude, labelField,
+                                          null);
+      SparkUtils.validateLabelFieldForTrainer(inputSchema, labelField);
+    }
+  }
+
+  public SparkMLTrainer(MLTrainerConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    pipelineConfigurer.createDataset(config.fileSetName, FileSet.class);
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    Preconditions.checkArgument(inputSchema != null, "Input Schema must be a known constant.");
+    config.validate(inputSchema);
+  }
+
+  @Override
+  public void prepareRun(SparkPluginContext context) throws Exception { }
+
+  @Override
+  public void run(SparkExecutionPluginContext context, JavaRDD<StructuredRecord> input) throws Exception {
+    if (!input.isEmpty()) {
+      final Schema inputSchema = input.first().getSchema();
+      final Map<String, Integer> featuresList = SparkUtils.getFeatureList(inputSchema, config.featureFieldsToInclude,
+                                                                          config.featureFieldsToExclude,
+                                                                          config.labelField);
+      JavaRDD<LabeledPoint> trainingData = input.map(new Function<StructuredRecord, LabeledPoint>() {
+        @Override
+        public LabeledPoint call(StructuredRecord record) throws Exception {
+          List<Double> featureList = new ArrayList<>();
+          List<Integer> featureIndex = new ArrayList<>();
+          int counter = 0;
+          for (String field : featuresList.keySet()) {
+            if (record.get(field) != null) {
+              featureList.add(((Number) record.get(field)).doubleValue());
+              featureIndex.add(counter);
+            }
+            counter++;
+          }
+          Double prediction = record.get(config.labelField);
+          if (prediction == null) {
+            throw new IllegalArgumentException(String.format("Value of Label field %s value must not be null.",
```

Comment:
Is it the intend that if any input is invalid (missing label field), the whole Spark job will fail? Or shall we skip the invalid input since it is invalid, it can't affect the prediction anyway.